app/src/main/java/com/example/development/sakaiclient20/ui/fragments/AllCoursesFragment.java

```diff
@@ -0,0 +1,173 @@
+package com.example.development.sakaiclient20.ui.fragments;
+
+import android.arch.lifecycle.ViewModelProviders;
+import android.content.Context;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
+import android.support.v4.widget.SwipeRefreshLayout;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.example.development.sakaiclient20.R;
+import com.example.development.sakaiclient20.models.Term;
+import com.example.development.sakaiclient20.networking.utilities.SharedPrefsUtil;
+import com.example.development.sakaiclient20.persistence.entities.Course;
+import com.example.development.sakaiclient20.ui.helpers.RutgersSubjectCodes;
+import com.example.development.sakaiclient20.ui.listeners.TreeViewItemClickListener;
+import com.example.development.sakaiclient20.ui.viewholders.CourseHeaderViewHolder;
+import com.example.development.sakaiclient20.ui.viewholders.TermHeaderViewHolder;
+import com.example.development.sakaiclient20.ui.viewmodels.CourseViewModel;
+import com.example.development.sakaiclient20.ui.viewmodels.ViewModelFactory;
+import com.unnamed.b.atv.model.TreeNode;
+import com.unnamed.b.atv.view.AndroidTreeView;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import dagger.android.support.AndroidSupportInjection;
+
+public class AllCoursesFragment extends Fragment {
+
+    public interface OnCourseSelectedListener {
+        void onCourseSelected(String siteId);
+    }
+
+    @Inject ViewModelFactory viewModelFactory;
+    private List<List<Course>> courses;
+    private AndroidTreeView treeView;
+    private OnCourseSelectedListener courseSelectedListener;
+
+    public static AllCoursesFragment newInstance(
```

Comment:
How come you are using a static newInstance method instead of a regular constructor?

Reply:
Was replying from email and didn't realize that the response would be to the PR and not this thread:

Two reasons:
1. https://stackoverflow.com/questions/9245408/best-practice-for-instantiating-a-new-android-fragment
discusses why overloading a Fragment constructor is bad practice (since
it's really easy to get it wrong)
2. The OnCourseSelectedListener cannot be bundled, as far as I'm aware,
since it's not a standard serializable object (I think, correct me if I'm
wrong on this)